pisces/models.py

```diff
@@ -72,69 +155,25 @@ def __init__(self,
-              ):
-        """
-        Assumes data is already preprocessed using `get_needed_X_y` 
-        and ready to be passed to the model.
+# %% ../nbs/02_models.ipynb 13
+class RandomForest(SleepWakeClassifier):
+    """Interface for sklearn's RandomForestClassifier"""
+    def __init__(self,
+                 data_processor: DataProcessor,
+                 class_weight: str = 'balanced',
+                 **kwargs):
+        self.model = RandomForestClassifier(class_weight=class_weight,
+                                            **kwargs)
+        self.scaler = StandardScaler()
+        self.pipeline = Pipeline([('scaler', self.scaler), ('model', self.model)])
+        if not isinstance(data_processor.model_input, ModelInput1D):
+            raise ValueError("Model input must be set to 1D on the data processor")
+        self.data_processor = data_processor
 
-        Returns the loss history of the model.
-        """
-        if (examples_X and not examples_y) or (examples_y and not examples_X):
-            raise ValueError("If providing examples, must provide both X and y")
-        else:
-            if examples_X and examples_y:
-                assert len(examples_X) == len(examples_y)
-        if pairs_Xy:
-            assert not examples_X
-            examples_X = [pair[0] for pair in pairs_Xy]
-            examples_y = [pair[1] for pair in pairs_Xy]
-
-
-        Xs = np.concatenate(examples_X, axis=0)
-        ys = np.concatenate(examples_y, axis=0)
-
-        selector = ys >= 0
-        Xs = Xs[selector]
-        ys = ys[selector]
-
-        loss_list = []
-        old_stdout = sys.stdout
-        sys.stdout = mystdout = StringIO()
-        self.pipeline.fit(Xs, ys) # Fit the model
-        sys.stdout = old_stdout
-        loss_history = mystdout.getvalue()
-        # Get loss
-        try:
-            for line in loss_history.split('\n'):
-                if(len(line.split("loss: ")) == 1):
-                    continue
-                loss_list.append(float(line.split("loss: ")[-1]))
-        except:
-            warnings.warn("Failed to fetch loss history. Returning empty list.")
-
-        return loss_list
-    
-    def _input_preprocessing(self, X: np.ndarray) -> np.ndarray:
-        return self.scaler.transform(X)
-    
-    def predict(self, sample_X: np.ndarray | pl.DataFrame) -> np.ndarray:
-        """
-        Assumes data is already preprocessed using `get_needed_X_y`
-        """
-        return self.model.predict(self._input_preprocessing(sample_X))
-    
-    def predict_probabilities(self, sample_X: np.ndarray | pl.DataFrame) -> np.ndarray:
-        """
-        Assumes data is already preprocessed using `get_needed_X_y`
-        """
-        return self.model.predict_proba(self._input_preprocessing(sample_X))
+    def get_needed_X_y(self, id: str) -> Tuple[np.ndarray, np.ndarray] | None:
+        return self.data_processor.get_1D_X_y(id)
 
-# %% ../nbs/02_models.ipynb 10
+# %% ../nbs/02_models.ipynb 15
```

Comment:
Let's add a `**kwargs` to the `init` of `MOResUNetPretrained`, and make the rest of the args (that make sense) have parallels there. eg instead of `tf_model` let's make it `model`.

This will help smooth out some of the switches on subclass type elsewhere.

Reply:
This makes sense! I'm also passing now `data_processor` to the init of MO, but this change will have effects downstream, on all our analysis scripts because now most of the functions don't require us to pass `data_processor` and rather get it from `self`.